langlib/lang.regexp/src/main/java/org/ballerinalang/langlib/regexp/Find.java

```diff
@@ -89,4 +96,22 @@ public static BArray findAllGroups(BRegexpValue regExp, BString str, int startIn
         }
         return groupArray;
     }
+
+    private static void checkIndexWithinRange(BString str, long startIndex) {
+        if (startIndex != (int) startIndex) {
+            throw BLangExceptionHelper.getRuntimeException(BallerinaErrorReasons.REGEXP_OPERATION_ERROR,
+                    RuntimeErrors.INDEX_NUMBER_TOO_LARGE, startIndex);
+        }
+
+        int strLength = str.length();
```

Comment:
```suggestion
```

Reply:
If it returns in the successive if; no need to get the length

Reply:
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/39238/commits/c68dce049bfb41f12e1fdade8ea90598b8bba533.